### PR TITLE
Core/Conditions: Added CONDITION_SOURCE_TYPE_TRAINER_SPELL

### PR DESCRIPTION
**Pull request by meji46**
**Changes proposed:**

-  Added CONDITION_SOURCE_TYPE_TRAINER_SPELL
-  Added missing AreaTriggerConditionConditionContainerStore deletion when reloading conditions

**Issues addressed:**
None

**Tests performed:**
Builds, tested in-game.

**Known issues and TODO list:** (add/remove lines as needed)
None

### CODE REVIEW

File: src/server/game/Entities/Creature/Trainer.cpp [hunk 1]
```diff
@@ -49,6 +50,12 @@ namespace Trainer
             if (!player->IsSpellFitByClassAndRace(trainerSpell.SpellId))
                 continue;
 
+            if (!sConditionMgr->IsObjectMeetingTrainerSpellConditions(_id, trainerSpell.SpellId, player))
+            {
+                TC_LOG_DEBUG("condition", "SendSpells: conditions not met for trainer id %u spell %u", _id, trainerSpell.SpellId);
```
Comment (Shauren) [inline src/server/game/Entities/Creature/Trainer.cpp:55]: You may want to include player in the log too

**Comment by Shauren** [comment]
Btw this is needed also for profession trainers in oribos (legendary item stuff is locked/hidden behind quests)

File: src/server/game/Conditions/ConditionMgr.h [hunk 2]
```diff
@@ -172,7 +172,8 @@ enum ConditionSourceType
     CONDITION_SOURCE_TYPE_AREATRIGGER                    = 28,
     CONDITION_SOURCE_TYPE_CONVERSATION_LINE              = 29,
     CONDITION_SOURCE_TYPE_AREATRIGGER_CLIENT_TRIGGERED   = 30,
-    CONDITION_SOURCE_TYPE_MAX                            = 31  // MAX
+    CONDITION_SOURCE_TYPE_TRAINER                        = 31,
```
Comment (Shauren) [inline src/server/game/Conditions/ConditionMgr.h:175]: Name is CONDITION_SOURCE_TYPE_TRAINER_SPELL (you are filtering individual spells, not access to trainer)

Comment (meji46) [reply src/server/game/Conditions/ConditionMgr.h]: Updated. I've also changed the name of the container

**Pull request merged**